### DISCUSSION TITLE
Feature suggestion: Given an assistant reply, let users write a prompt for it

### DISCUSSION THREAD

**Discussion by erkinalp** [LAION-AI/Open-Assistant#3446] [Ideas]
This training scenario might be useful to train latent diffusor (http://arxiv.org/abs/2212.09462) and combined diffusor+GAN architectures of generative models.

**Comment by mm12** [comment]
interesting idea, like that one game show i'm not gonna name. I can see the potential benefits.

**Comment by horribleCodes** [comment]
By prompt, do you mean the prompt that would generate the reply? I wonder what we could expect to see from that. I'm assuming the user would have to provide something that would work as an initial prompt, because "Now give me a list of these places." wouldn't be a very effective example. Then again, the assistant could be making a reference to an earlier statement, in which case the correct prompt would be a user reply - unless you limit the assistant messages to the first ones of each tree only.

**Comment by erkinalp** [comment]
> By prompt, do you mean the prompt that would generate the reply?

Yes

**Comment by michaelbogdan** [comment]
This could be used as a way to train and tune [doc2query](https://zbmath.org/software/37761) as well - there a somewhat related issue in this project with #645. Could also be used to connect a given response to more prompts and thereby increasing the dataset for the reward model to train on.

**Comment by huu4ontocord** [comment]
This is actually a brilliant idea @erkinalp. This will create diversity for the data. You would then be creating a new sister subtree tho. So this could complicated things. I will leave for the webteam to decide if this can be implemented. i adding appropriate labels to let that team know.

**Comment by horribleCodes** [comment]
Maybe the trees created from this should be in a different dataset altogether? That'd make things easier, but I don't know if there's a benefit in putting the predicted prompts in the same dataset.
